### PR TITLE
Feat(result logs): Use three logging levels

### LINKED ISSUE #2680: FreshnessStatus.Error causes exit status of 1 but no error level logging.
### Describe the bug
When I run `dbt --log-format=json source snapshot-freshness` and there is a `SourceFreshnessResult` with  `status = FreshnessStatus.Error: 'error'`, the log level of the output is `INFO` but message has a red `ERROR` and the cli returns with status code 1.

### Steps To Reproduce
As far as I can tell this will happen with any stale snapshot when you run `dbt --log-format=json source snapshot-freshness`.

### Expected behavior
I expect the log level of the output to be error.

### Screenshots and log output
```
{
  "timestamp": "2020-08-03T19:16:19.227088Z", 
  "message": "15:16:19 | 1 of 1 ERROR STALE freshness of foo.bar.............. [\u001b[31mERROR STALE\u001b[0m in 2.43s]", 
  "channel": "dbt", "level": 11,
  "levelname": "INFO",    <------------ ???
  "thread_name": "Thread-1", 
  "process": 65779, 
  "extra": {"unique_id":  "source.db.sch.tab", "run_state": "running"}
}
```

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [x] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
installed version: 0.16.1
   latest version: 0.17.2
```

**The operating system you're using:** MacOS Catalina

**The output of `python --version`:** 3.8.1

### Additional context
We parse the json output of a dbt run and report any ERROR level log statements to a monitoring tool, so this causes us to have a gap in our reporting.

Please let me know if I've mistaken a feature for a bug. Happy to contribute a fix if it is something that needs fixing.

### PR DESCRIPTION
**Pull request by tpilewicz**
resolves #2680

### Description

I've added an argument `logger_fn` to the function `print_fancy_output_line` of module `printer.py`, that determines the logging level used to print logs. Then I've used it accordingly where the function is called. Also, I've modified the function `get_printable_result` from this module to determine a `logger_fn` from model / seed / snapshot results.

I have not added tests because this module was not tested, and I don't really know how to test logging functions yet. Tell me if you think it is better to add some.

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

**Comment by jtcohen6** [comment]
Thanks for this contribution, @tpilewicz!

I'm going to hold off on merging this into the `dev/marian-anderson` branch, as we're currently in release-candidate phase and will be releasing v0.18.0 very soon.

This is a good change; what's less clear to me is whether it'd be a breaking change for users who deploy dbt via orchestration tools, such as Airflow, that have native behavior corresponding to log levels. Should we wait to merge this until the next minor version (0.19), or would it be okay to include in a patch release? @beckjake I'd be curious to hear your thoughts here.

**Comment by beckjake** [comment]
Well, first of all, does this actually have any significant change in log level, in the sense of "an error was logged"? I assume that's what airflow looks at.

This adds an error log if there is an `ERROR SKIP` (if this doesn't log an error already, that seems like a bug!) or if the run is canceled with SIGINT.  I think I'm fine with it if this change happens to break someone's airflow environment by causing those two cases to fail when they previously didn't.

I think saying that log levels can only be changed in minor/major revs is a really inconvenient precedent to set. Where does it end? Can we rev dependencies? Change print output? Change error handling?

**Comment by jtcohen6** [comment]
> I think saying that log levels can only be changed in minor/major revs is a really inconvenient precedent to set. Where does it end? Can we rev dependencies? Change print output? Change error handling?

Cool, let's pull this into v0.18.1.

**Comment by tpilewicz** [comment]
Great, thanks for your review guys 😄

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@
 - Macros in the current project can override internal dbt macros that are called through `execute_macros`. ([#2301](https://github.com/fishtown-analytics/dbt/issues/2301), [#2686](https://github.com/fishtown-analytics/dbt/pull/2686))
 - Add state:modified and state:new selectors ([#2641](https://github.com/fishtown-analytics/dbt/issues/2641), [#2695](https://github.com/fishtown-analytics/dbt/pull/2695))
 - Add two new flags `--use-colors` and `--no-use-colors` to `dbt run` command to enable or disable log colorization from the command line ([#2708](https://github.com/fishtown-analytics/dbt/pull/2708))
+- Specify all three logging levels (`INFO`, `WARNING`, `ERROR`) in result logs for commands `test`, `seed`, `run`, `snapshot` and `source snapshot-freshness`
```
Comment (jtcohen6) [inline CHANGELOG.md:33]:
```suggestion
- Specify all three logging levels (`INFO`, `WARNING`, `ERROR`) in result logs for commands `test`, `seed`, `run`, `snapshot` and `source snapshot-freshness` ([#2680](https://github.com/fishtown-analytics/dbt/pull/2680), [#2723](https://github.com/fishtown-analytics/dbt/pull/2723))
```

Could you:
* merge changelog conflicts
* move this entry to v0.18.1
* create a Contributors section
* add yourself!

**Review by jtcohen6 — APPROVED** [review]
Looks + works great! Thanks for the contribution @tpilewicz

**Review by jtcohen6 — APPROVED** [review]

**Pull request merged**